task_processing/interfaces/event.py

```diff
@@ -1,15 +1,20 @@
 from pyrsistent import field
+from pyrsistent import m
+from pyrsistent import PMap
 from pyrsistent import PRecord
 
-# TODO: organize and explain these
+from task_processing.interfaces.task_executor import TaskConfig
 
 
 class Event(PRecord):
```

Comment:
`middleware_data` is an attempt to constrain the growth. we definitely need this record to be extensible without having to modify the source.